packages/pluggableWidgets/switch-web/src/__tests__/Switch.spec.tsx

```diff
@@ -0,0 +1,329 @@
+
+            expect(props.action?.execute).toHaveBeenCalled();
+        });
+
+        it("should not invoke action on keydown of any key but space", () => {
+            const props = createProps({ action: actionValue() });
+            createAndFindElements(props);
+
+            switchButtonWrapper.simulate("keydown", { key: "enter" });
+
+            expect(props.action?.execute).not.toHaveBeenCalled();
+        });
+
+        describe("when value is available", () => {
+            it("should toggle the attributes value onClick", () => {
+                const props = createProps({
+                    booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
+                });
+                createAndFindElements(props);
+
+                switchButton.simulate("click");
+
+                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
+                expect(props.booleanAttribute.value).toEqual(true);
+
+                switchButton.simulate("click");
+
+                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
+                expect(props.booleanAttribute.value).toEqual(false);
+            });
+
+            it("should toggle the attribute value on space keydown", () => {
+                const props = createProps({
+                    booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
+                });
+                createAndFindElements(props);
+
+                switchButton.simulate("keydown", { key: " " });
+
+                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
+                expect(props.booleanAttribute.value).toEqual(true);
+
+                switchButton.simulate("keydown", { key: " " });
+
+                expect(props.booleanAttribute.setValue).toHaveBeenCalled();
+                expect(props.booleanAttribute.value).toEqual(false);
+            });
+
+            it("should not toggle the attribute on keydown of any key but space", () => {
+                const props = createProps({
+                    booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).build()
+                });
+                createAndFindElements(props);
+
+                switchButton.simulate("keydown", { key: "enter" });
+
+                expect(props.booleanAttribute.setValue).not.toHaveBeenCalled();
+            });
+        });
+
+        describe("when value is loading", () => {
+            it("should not toggle the attributes value", () => {
+                const props = createProps({
+                    booleanAttribute: new EditableValueBuilder<boolean>().withValue(false).isLoading().build()
+                });
+                createAndFindElements(props);
+
+                switchButton.simulate("click");
+
+                expect(props.booleanAttribute.setValue).not.toHaveBeenCalled();
+            });
+
+            it("should not toggle the attributes value on space keydown", () => {
```

Comment:
Isn't this being guaranteed by the APIs?

Reply:
It is, thanks. I was over cautious here 😊